public/app/features/alerting/unified/components/rule-editor/AnnotationsField.tsx

```diff
@@ -59,76 +76,133 @@ const AnnotationsField = () => {
-                      existingKeys={existingKeys(index)}
-                      width={18}
-                    />
-                  )}
-                  control={control}
-                  rules={{ required: { value: !!annotations[index]?.value, message: 'Required.' } }}
-                />
-              </Field>
-              <Field
-                className={cx(styles.flexRowItemMargin, styles.field)}
-                invalid={!!errors.annotations?.[index]?.value?.message}
-                error={errors.annotations?.[index]?.value?.message}
-              >
-                <ValueInputComponent
-                  data-testid={`annotation-value-${index}`}
-                  className={cx(styles.annotationValueInput, { [styles.textarea]: !isUrl })}
-                  {...register(`annotations.${index}.value`)}
-                  placeholder={isUrl ? 'https://' : `Text`}
-                  defaultValue={annotationField.value}
-                />
-              </Field>
-              <Button
-                type="button"
-                className={styles.flexRowItemMargin}
-                aria-label="delete annotation"
-                icon="trash-alt"
-                variant="secondary"
-                onClick={() => remove(index)}
-              />
+              <div>
+                <div>
+                  <label className={styles.annotationContainer}>
+                    {
+                      <InputControl
+                        name={`annotations.${index}.key`}
+                        defaultValue={annotationField.key}
+                        render={({ field: { ref, ...field } }) =>
+                          annotationField.key === Annotation.dashboardUID ? (
+                            <div>Dashboard and panel</div>
+                          ) : annotationField.key === Annotation.panelID ? (
+                            <span></span>
+                          ) : (
+                            <div>
+                              <span className={styles.annotationTitle} data-testid={`annotation-key-${index}`}>
+                                {annotationLabels[annotation]}{' '}
+                              </span>
+                              {annotationLabels[annotation] ? (
+                                <span className={styles.annotationTitle}>(optional)</span>
+                              ) : (
+                                <div>
+                                  <span className={styles.annotationTitle}>Custom annotation name and content</span>
+                                  <Input
+                                    placeholder="Enter custom annotation name..."
+                                    width={18}
+                                    {...field}
+                                    className={styles.customAnnotationInput}
+                                  />
+                                </div>
+                              )}
+                            </div>
```

Comment:
I would wrap this code as a separate component and make the conditionals more clear

Reply:
Good idea, done